stackslib/src/net/connection.rs

```diff
@@ -415,6 +415,8 @@ pub struct ConnectionOptions {
     /// the reward cycle in which Nakamoto activates, and thus needs to run both the epoch
     /// 2.x and Nakamoto state machines.
     pub force_nakamoto_epoch_transition: bool,
+    /// The authorization password to enable the block proposal RPC endpoint
+    pub supports_block_proposals: Option<String>,
```

Comment:
I think the var name could be a little better, maybe something like `block_proposal_token`? Or something more generic if we think it could be useful elsewhere?

I get that this field does two things - enable the endpoint, and then specify the token, so it is tricky.

Reply:
Seconded